md: TestRenderCodeBlock fails — closing fence not preceded by newline

## Summary

`TestRenderCodeBlock` in `md/md_renderer_test.go` fails on `master` (and has apparently been failing for a long time): the `codeBlock` renderer writes the closing `` ``` `` fence directly after the node literal, so for a `CodeBlock` whose `Literal` does not end with `\n` the output becomes invalid markdown with the closing fence glued to the last content line.

This goes unnoticed because the CI workflow (`.github/workflows/go.yml`) runs `go test` only on `.`, `./ast`, `./parser`, `./html` — the `./md` package is not in the list.

## Reproduce

On the current `master` (tested at `134a5b2`):

```
go test -v -run TestRenderCodeBlock ./md
```

```
--- FAIL: TestRenderCodeBlock
    md_renderer_test.go:184: [
        ```scala
        val x : Int = 42```

        ] is not equal to [
        ```scala
        val x : Int = 42
        ```
        ]
FAIL    github.com/gomarkdown/markdown/md
```

## Root cause

`md/md_renderer.go`, `codeBlock`:

```go
r.outs(w, "\n")
r.out(w, text)
r.outs(w, "```\n\n")
```

If `text` does not already end with `\n`, the closing fence ends up on the same line as the content.

For `CodeBlock` nodes produced by the parser the `Literal` typically already ends with a newline, so the bug is not visible in `TestRenderCodeWithParagraph` (which feeds the renderer a parsed input). It reproduces when the AST is constructed manually, which is exactly what `TestRenderCodeBlock` does.

## Suggested fix

Emit `\n` before the closing fence when `text` does not already end with one; and update `TestRenderCodeBlock`'s expected string to end with `\n\n` (matching the `doubleSpace` + closing-block semantics of every other test in the file):

```go
// md/md_renderer.go
r.outs(w, "\n")
r.out(w, text)
if len(text) == 0 || text[len(text)-1] != '\n' {
    r.outs(w, "\n")
}
r.outs(w, "```\n\n")
```

```go
// md/md_renderer_test.go
- expected := "\n```scala\nval x : Int = 42\n```\n"
+ expected := "\n```scala\nval x : Int = 42\n```\n\n"
```

With this change all 16 tests in `./md` pass; other packages are unaffected.

Happy to open a PR with the fix.

## Additional note

It may be worth extending the `Test` step in `.github/workflows/go.yml` to include `./md` (or just use `go test ./...` with `examples/` excluded somehow) so regressions in this package surface automatically.

--- a/md/md_renderer.go
+++ b/md/md_renderer.go
@@ -264,6 +264,9 @@ func (r *Renderer) codeBlock(w io.Writer, node *ast.CodeBlock) {
 	}
 	r.outs(w, "\n")
 	r.out(w, text)
+	if len(text) == 0 || text[len(text)-1] != '\n' {
+		r.outs(w, "\n")
+	}
 	r.outs(w, "```\n\n")
 }
 

--- a/md/md_renderer_test.go
+++ b/md/md_renderer_test.go
@@ -73,7 +73,7 @@ func TestRenderCode(t *testing.T) {
 func TestRenderCodeBlock(t *testing.T) {
 	input := &ast.CodeBlock{Info: []byte(string("scala"))}
 	input.Literal = []byte(string("val x : Int = 42"))
-	expected := "\n```scala\nval x : Int = 42\n```\n"
+	expected := "\n```scala\nval x : Int = 42\n```\n\n"
 	testRendering(t, input, expected)
 }
 

